src/domain/usecases/SetSkipOfflineDataSetUseCase.ts

```diff
@@ -0,0 +1,55 @@
+import _ from "lodash";
+
+import { DataSetsRepository } from "../repositories/DataSetsRepository";
+import { Maybe } from "../../utils/ts-utils";
+import { DataSet } from "../entities/DataSet";
+import log from "../../utils/log";
+
+type SetSkipOfflineOptions = {
+    year: number;
+    disable: Maybe<boolean>;
+};
+
+export class SetSkipOfflineDataSetUseCase {
+    constructor(private dataSetRepository: DataSetsRepository) {}
+
+    async execute(options: SetSkipOfflineOptions) {
+        const updatedDataSets = await this.dataSetRepository
```

Comment:
Adding a log like 'Loading dataSets...' here could be useful, since fetching all the dataSets from the server might take a long time.